untitled1/src/test/com/aca/homework/week8/credit/card/PaymentServiceTest.java

```diff
@@ -0,0 +1,34 @@
+package com.aca.homework.week8.credit.card;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.Executable;
+
+class PaymentServiceTest {
+    @Test
+    public void testPayWhenBalanceLessThenAmountToBePaid() {
+        InsufficientBalanceException insufficientBalanceException = Assertions.assertThrows(InsufficientBalanceException.class, new Executable() {
+            @Override
+            public void execute() throws Throwable {
+                PaymentService.pay(new CreditCard(100, "12457"), 150);
+            }
+        });
+        Assertions.assertEquals("the provided credit card balance: 100.0 is not enough to perform a payment equals 150.0" , insufficientBalanceException.getMessage());
+    }
+
+    @Test
+    public void testPayWhenBalanceGreaterThenAmountToBePaid() {
+        InsufficientBalanceException insufficientBalanceException = Assertions.assertThrows(InsufficientBalanceException.class, new Executable() {
+            @Override
+            public void execute() throws Throwable {
```

Comment:
you can not identify at which pay method the exception was thrown. I suggest to invoke 3 pay methods out of the execute and 4th one here